src/Dialog/Dialog.test.tsx

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2015 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
+ */
+
+import React from 'react';
+import { fireEvent, screen } from '@testing-library/react';
+
+import render from '../../test/testrenderer';
+import {
+    ConfirmationDialog,
+    Dialog,
+    DialogButton,
+    ErrorDialog,
+    InfoDialog,
+} from './Dialog';
+
+const noop = jest.fn();
+describe('Dialog', () => {
+    const dialog = (isVisible = true) => (
+        <Dialog isVisible={isVisible}>
+            <Dialog.Header title="Test Title" />
+            <Dialog.Body>Test Text</Dialog.Body>
+            <Dialog.Footer>
+                <DialogButton onClick={noop}>Close</DialogButton>
+            </Dialog.Footer>
+        </Dialog>
+    );
+
+    describe('is visible when it', () => {
+        it('is rendered when visible', () => {
+            render(dialog());
+            expect(screen.getByRole('dialog')).toBeInTheDocument();
+            test;
+        });
+
+        it('is not rendered when invisible', () => {
+            render(dialog(false));
+            expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+        });
+    });
+
+    describe('shows the expected content', () => {
+        it('in the title', () => {
+            render(dialog());
+
+            expect(screen.getByText('Test Title')).toBeInTheDocument();
+        });
+
+        it('in the body', () => {
+            render(dialog());
+
+            expect(screen.getByText('Test Text')).toBeInTheDocument();
+        });
+
+        it('in the footer', () => {
+            render(dialog());
+
+            expect(screen.getByText('Close')).toBeInTheDocument();
+        });
+    });
+});
```

Comment:
These can all be a single test. Suggest to remove the describe and create a single test test('contains the expected content', () => {});

Reply:
Fixed